components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1603,6 +1603,25 @@ _resolve_dependencies() {
   _validate_deps
 }
 
+function do_automatic() {
+  do_default_automatic "$1"
+  return $?
+}
+
+# Loads specified automatic to extend plan DSL.
+function do_default_automatic() {
+  local automatic="${automatic:-$1}"
+
+  if [[ -n "$automatic" ]]; then
+    build_line "Loading Automatic $automatic"
+    if source "$(pkg_path_for $automatic)/bin/automatic.sh"; then
```

Comment:
I was thinking about the location for loading this file…it might make more sense to pull this out of the `PATH` of the blueprint package. Perhaps putting it under `lib/automatic.sh` (or even `libexec/automatic.sh`) that way if there were ever more than one, we wouldn't have (potentially) multiple `automatic.sh` files showing in `$PATH`. As it's a plugin of sorts it probably makes sense to think of it as a library implementation (hence why I was thinking putting it under `lib/`)